### ISSUE TITLE
atmos list should be able to filter for real and abstract components

### ISSUE THREAD

**Issue by astephanh** [cloudposse/atmos#833]
### Describe the Feature

since atmos version 1.110 `atmos list' is implented instead of a custom command.  

This is a good thing, as it is much faster then using the custom command.

But It can no longer be overwritten using atmos.yaml custom command. For me it now lacks the functionality to filter for real or abstract components.

### Expected Behavior

add functionality to filter for real / abstract components

### Use Case

if you have a lot of abstract components, filtering is important

### Describe Ideal Solution

add flag --type / -t to filter for abstract / real components

**Comment by osterman** [comment]
@astephanh agreed! In fact, we've already assigned a task to work on this. We'll link to this issue when the PR is opened.

**Comment by RoseSecurity** [comment]
Is it possible to expand this to listing enabled and disabled components?

**Comment by mss** [comment]
@RoseSecurity I think that makes sense. It should probably default to hiding disabled components.

Unfortunately was the referenced change #828 stalled; it included a lot of not really related stuff as well though. I poked at it a bit if I can extract the changes which are relevant for this issue but didn't have much time to do so. I just noticed that you contributed the original `list components` feature so maybe you are more familiar with the codebase? :-)

**Comment by osterman** [comment]
We have an up coming task to implement this, so we will ensure the requirements include also being able to filter by enabled/disabled, locked and unlocked, and any combination of that. In addition to the original request to filter for abstract and real.
